### DISCUSSION TITLE
Granting Accessibility Access

### DISCUSSION THREAD

**Discussion by noah-nuebling** [noah-nuebling/mac-mouse-fix#101] [Guides]
To use Mac Mouse Fix you'll have to grant it 'Accessibility Access'.
In this Guide you'll learn about common issues with granting Accessibility Access and how to solve them.

> [!NOTE]
> **If you're on macOS 10.14 Mojave or older** 
> Please refer to this guide for solutions.
>
>  **If you're on macOS 10.15 Catalina or newer** 
> Mac Mouse Fix 2.2.2 and later should automatically resolve the issues described here. \
> If you experience problems despite this, please leave a comment below, so I can help you out and update this guide. Thank you!

---

### 1. Issue where Accessibility Access is enabled but Mac Mouse Fix doesn't recognize that

Sometimes 'Mac Mouse Fix Helper' will show up as enabled in the list at 'System Settings > Privacy & Security > Accessibility' but Mac Mouse Fix will keep asking you to grant Accessibility Access.

<img width="600" align="center" src="https://user-images.githubusercontent.com/40808343/198885178-a321eabd-308a-4b23-865d-1bd475c79cd5.png">

#### Why does this issue occur?

I'm not sure but it seems that sometimes the macOS Accessibility Access database doesn't update properly after you update Mac Mouse Fix. 
See [this GitHub Issue](https://github.com/noah-nuebling/mac-mouse-fix/issues/412) for more information.

#### Solution

If you encounter this issue, you can still grant Accessibility Access like so:

https://user-images.githubusercontent.com/40808343/198886682-cfbe7832-7e4f-4434-8b51-e4ff89c720f5.mp4

1. Go to 'System Settings > Privacy & Security > Accessibility'
   - You can get there quickly by clicking on "Open 'Privacy & Security' Settings" in Mac Mouse Fix
   - In the video, I then click on the 'Mac Mouse Fix Helper' toggle repeatedly. This is to demonstrate that the toggle doesn't work, it is not a neccessary step of the solution.
2. If you're using macOS 12 Monterey or older, click the lock in the lower left and then enter your password to unlock System Preferences
3. Delete the entry for 'Mac Mouse Fix Helper' from the Accessibility list by selecting it and then clicking the minus (-) button at the bottom of the list
4. Go back to the Mac Mouse Fix window, then click the Mac Mouse Fix Icon while holding the Shift (⇧) key to reveal 'Mac Mouse Fix Helper' in the Finder
5. Drag and drop 'Mac Mouse Fix Helper' from the Finder to the Accessibility List in System Preferences
6. You should now be able to use Mac Mouse Fix!

---

### 2. Issue where 'Mac Mouse Fix Helper' won't show up in System Preferences

Sometimes, 'Mac Mouse Fix Helper' doesn't show up in the list at 'System Settings > Privacy & Security > Accessibility' preventing you from granting Accessibility Access to it.

<img width="400" align="center" src="https://user-images.githubusercontent.com/40808343/198889194-195abb41-3341-4563-b72c-97e29c2da4f5.png">

#### Why does this issue occur?

I'm not sure why this issue occurs. I assume that it's a bug in macOS. If you have any ideas what else might cause it let me know!

#### Solution
If you encounter this issue, you can still grant Accessibility Access like this:

https://user-images.githubusercontent.com/40808343/198889288-88058924-c190-4bf3-955b-1e9e988bf21c.mp4

1. Go to 'System Settings > Privacy & Security > Accessibility'
   - You can get there quickly by clicking on "Open 'Privacy & Security' Settings" in Mac Mouse Fix
2. If you're using macOS 12 Monterey or older, click the lock in the lower left and then enter your password to unlock System Preferences
3. Go back to the Mac Mouse Fix window, then click the Mac Mouse Fix Icon while holding the Shift (⇧) key to reveal 'Mac Mouse Fix Helper' in the Finder
4. Drag and drop 'Mac Mouse Fix Helper' from the Finder to the Accessibility List in System Preferences
5. You should now be able to use Mac Mouse Fix!

---

### Notes on changes to System Settings under macOS 13 Ventura

Apple rewrote the 'System Preferences' app for macOS 13 Ventura and renamed it to 'System Settings'. This leads to a few caveats for this guide:

- In several places the guide refers to a the list at 'System Settings > Privacy & Security > Accessibility'. On macOS 12 Monterey and below you can find this list under 'System Preferences > Security & Privacy > Privacy > Accessibility' instead.
- The video footage and images for this Guide stem from the new System Settings introduced in macOS 13 Ventura. Things will look a little different under macOS 12 Monterey and below but the same issues can occur and the same solutions apply.

---

I hope this Guide helped you! If you have any questions you can:
- Write a comment down below
- [Open a new GitHub Discussion](https://github.com/noah-nuebling/mac-mouse-fix/discussions)
- [Send me an Email](mailto:noah.n.public@gmail.com?)
  - Please note that I get many emails and I don't have that much time, so I might not be able to answer

General feedback and improvement ideas for this guide are also very welcome of course!

**Comment by tonilampela** [comment]
Same issue in Monterey too after updating to Mac Mouse Fix 2.2.1.

**Reply by noah-nuebling** [reply]
Hey @tonilampela! See [this comment](https://github.com/noah-nuebling/mac-mouse-fix/discussions/101#discussioncomment-3937827).

**Comment by robertburke14** [comment]
Yep, just updated, running Monterey, it's switched on in settings but keeps prompting me to turn it on. "Have you tried turning it off then on again?" Yes, and turning it off froze my machine and I had to force reboot, at which point it reset my trackpad speed for some reason. I then switched it back on, and still am getting the message to turn it on from Mac Mouse Fix.

**Reply by noah-nuebling** [reply]
Hi! Sorry for the trouble. 

This issue should be fixed in the [latest update](https://github.com/noah-nuebling/mac-mouse-fix/releases)!
See the top of [this issue](https://github.com/noah-nuebling/mac-mouse-fix/issues/412#issue-1415384638) for more info.

Hope that helps! Let me know if you have more questions.

**Comment by ADGArrio** [comment]
I just updated the app and although I have given it accessibility access in System Preferences it still tell me to give it access. Please fix this bug; I love this app.

**Reply by noah-nuebling** [reply]
Hey @ADGArrio, thanks for you feedback! :)

See [this comment](https://github.com/noah-nuebling/mac-mouse-fix/discussions/101#discussioncomment-3937827)

**Comment by FrankKair** [comment]
Same issue running `2.2.1` on _Monterey 12.6_.

**Reply by noah-nuebling** [reply]
Hey @FrankKair, see [this comment](https://github.com/noah-nuebling/mac-mouse-fix/discussions/101#discussioncomment-3937827)!

**Reply by FrankKair** [reply]
Updated to `2.2.2` and it works ❤️ Thank you so much!

**Reply by noah-nuebling** [reply]
Great! Thanks for getting back!

**Comment by xdtxAM** [comment]
## The latest version doesn't work properly with macOS 13.0, it always reminds me to open Privacy Setting, but I'm pretty sure I have this option turned on in the system settings

**Reply by noah-nuebling** [reply]
Hey @gwaiting! See [this comment](https://github.com/noah-nuebling/mac-mouse-fix/discussions/101#discussioncomment-3937827).

**Comment by sashat** [comment]
My Monterrey will not record keyboard shortcuts with the new MacMouseFix version. Previously in the past I could record "Command + c" to mouse button 4 fine. According to the description this behavior was fixed for Mohave but seems it broke this for Monterrey :(

**Reply by noah-nuebling** [reply]
Oh hey, sorry for the (very) late reply! Is this still the case?

**Reply by sasa4uk** [reply]
2 years have passed, it is not the case anymore :) I am running different OS version as well as mac mouse fix. From what I remember then - I just switched to another version of MacMouseFix app to fix the issue. Obviously I can't remember which that was exactly. But thanks for asking anyways. The tool is great overall!

**Reply by noah-nuebling** [reply]
Thanks! :)

**Comment by BruceWayne426** [comment]
I am running a 2014 MacBook Pro with Big Sur 11.7.10. I have had a Logitech M525 mouse for years with two programmable buttons that I could never get to work. The mouse works but the Logitech drivers said the programmable buttons were only available on Windows computers. After discovering your program, I followed the directions in the video to enable your program access. I can now program my buttons. Thanks for a great fix!

**Comment by Chapps-SL** [comment]
I'm trying to solve the lack of support on a Magic Mouse for zoom within programs (including Preview). This of course works with a trackpad, with pinch to zoom, but now that Wacom no longer supports touch gestures on its new graphics tablet, I need a workaround. I've installed your app, but the keyboard modifiers don't work - none of them. Of course, that's were zoom is supported. Any workarounds?
